lib/src/api/bucket.dart

```diff
@@ -33,12 +27,24 @@ class Bucket {
     return File(this, key);
   }
 
+  $p.StorageClient _getClient() {
+    if (_storageClient != null) {
+      return _storageClient;
+    }
+
+    return $p.StorageClient(ClientChannelSingleton.instance.clientChannel);
+  }
+
   /// Get a list of references to the files in the bucket. Optionally supply a [prefix] to filter by.
   Future<List<File>> files({String prefix = ""}) async {
+    final storageClient = _getClient();
+
     final request =
         $p.StorageListBlobsRequest(bucketName: name, prefix: prefix);
 
-    var resp = await _storageClient.listBlobs(request);
+    var resp = await storageClient.listBlobs(request);
+
+    await ClientChannelSingleton.instance.release();
```

Comment:
the location where the channel singleton is leased and it is being released are a bit too separated in these methods, and the singleton is only leased in the case where a `_storageClient` wasn't provided in the constructor meaning that the methods could release channels being held elsewhere in error.

The client constructor could take a callback, that releases the channel after the method that needs to use the client is done?

e.g. psuedocode

```
getClient(callback) {
   var channel = ClientChannelSingleton.instance.clientChannel;
   callback($p.StorageClient(ClientChannelSingleton.instance.clientChannel))
   ClientChannelSingleton.instance.release();
}
```